Infra Password and SQL Password

LiliamLeme opened the pull request:
# Type of PR
<!-- Removed items that do not match with your change. -->- Code changes

1 - Deploy deploy_infrastructure
- Simplify the logic - If the character is Hypen
- replace by R. Instead of a retry logic

2 - Init_enviromnent:
- Added more complex string concatenated to
- SQL password


## Purpose
<!-- Describe the intention of the changes being proposed. What problem does it solve or functionality does it add? -->

1 - Deploy deploy_infrastructure
- Simplify the logic - If the character is Hypen
- replace by R. Instead of a retry logic

2 - Init_enviromnent:
- Added more complex string concatenated to
- SQL password

## Does this introduce a breaking change? If yes, details on what can break

## Author pre-publish checklist
<!-- Please check check before publishing PR using "x". Remove a column if it's not applicable. -->
- [ X] Added test to prove my fix is effective or new feature works
- [ X] No PII in logs
- [ ] Made corresponding changes to the documentation

## Validation steps

Testes log result example:
+ sp_stor_pass='Validating...-fm8....
Invalid password. The first character cannot be a hyphen. Replacing...
Rfm8....
Password Validated
+ echo 'ADLS - Valid password obtained'

## Issues Closed or Referenced
<!-- This will automatically close the issue when the PR closes. -->
- Closes #issue_number
<!-- this references the issue but does not close with PR. -->
- References #issue_number

thesqlpro commented on e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh:
```diff
@@ -204,40 +200,28 @@ validate_password() {
 
 # ###########################
-###if there is an hyphen retry and recreate the SP with a new password
+###if there is an hyphen in the first position replace by R
```
wording is a bit confusing in the comment
you mean replace with the letter R correct?
and should the comment be here, or in the section above (the function itself)?

thesqlpro commented on e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh:
```diff
@@ -204,40 +200,28 @@ validate_password() {
 echo "Creating Service Principal (SP) for access to ADLA Gen2 used in Databricks mounting"
```
ADLA should be ADLS

LiliamLeme replied:
ADLA was not me. It was there already when I arrived... but I fixed

thesqlpro reviewed (commented):
Just two comments to be addressed
Looks good!

LiliamLeme commented:
Removed the complexity of the deploy_infra to validate the password in case there is a Hyphen
re-tested
Added a complex string on top of the makepasswrd for SQL Server so it will be a always complex when generated.

LiliamLeme commented:
Removed the complexity of the deploy_infra to validate the password in case there is an Hyphen
re-tested
Added a complex string on top of the makepasswrd for SQL Server so it will be a always complex when generated.
